### PR TITLE
Testpatch

### PR DESCRIPTION
**Pull request by BelaPfeffer**
## Description

Implemented all changes suggested in review on Friday. Should be the final version of the Simde Implementation of the XML Parsing Problem. 

What we did: 

We tired to speed up the toNative method in the XMLHandler.
Therefore we changed the implementation of appendASCII. It now uses SIMDe to process the Input in a Block of 8 chars at a time.
We added two three new methods to the StringManager Class
1.)int StringManager::strLength(const XMLCh* input_ptr)
This method uses SIMDe commands to determine the length of a Xerces String by finding the 0 iterator
2.)void StringManager::compress64(const XMLCh* inputIt, char* outputIt)
This method helps the appendASCII method to compress 8 UTF 16 char at a time down to ASCII chars also using SIMDe
3.)bool StringManager::isASCII(const XMLCh* chars, const XMLSize_t length)
This Method determines whether or not an Input is ASCII convertible or not by checking the upper bytes of a UTF 16 char.
Underlying logic : very UTF -16 char with two zeros as upper bits is ASCII convertible.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

## Summary by CodeRabbit

- **Performance Improvements**
  - Enhanced string handling for XML data with SIMD acceleration, improving processing speed for ASCII and UTF-16 encoded strings.

- **Bug Fixes**
  - Improved accuracy and reliability in detecting and converting ASCII and non-ASCII characters in XML strings.

- **Tests**
  - Added comprehensive tests for string handling functions covering various character sets and edge cases.

- **Style**
  - Minor formatting and whitespace adjustments for improved code readability.

### CODE REVIEW

**Comment by cbielow** [comment]
thanks for the PR.
Please address the comments from CodeRabbit (see above).

File: src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h [hunk 1]
```diff
@@ -283,7 +299,11 @@ namespace OpenMS
       {
         return toNative_(str);
       }
+      /// Checks if supplied if chars in XMLCh* can be encoded with ASCII
```
Comment (cbielow) [inline src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h]:
```suggestion
      /// Checks if supplied chars in XMLCh* can be encoded with ASCII (i.e. the upper byte of each char is 0)
```

File: src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp [hunk 2]
```diff
@@ -302,7 +303,7 @@ namespace OpenMS::Internal
         }
         // no value, although there should be a numerical value
         else if (term.xref_type != ControlledVocabulary::CVTerm::NONE && term.xref_type != ControlledVocabulary::CVTerm::XSD_STRING && // should be numerical
-                 !cv.isChildOf(accession, "MS:1000513") // here the value type relates to the binary data array, not the 'value=' attribute!
+                 !cv.isChildOf(accession, "MS:1000513") // here the value type relates to the bits data array, not the 'value=' attribute!
```
Comment (cbielow) [inline src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp]: revert please

File: src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp [hunk 3]
```diff
@@ -417,43 +418,151 @@ namespace OpenMS::Internal
...
+      {
+        compress64_(inputPtr, outputPtr);
+        inputPtr += 8;
+        outputPtr += 8;
+      }
+    
+      // Copy any remaining characters individually
+      for (Size i = 0; i < remainder; ++i)
+      {
+        outputPtr[i] = static_cast<char>(inputPtr[i] & 0xFF);
```
Comment (cbielow) [inline src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp:553]:
```suggestion
        outputPtr[i] = static_cast<char>(inputPtr[i]);
```
